go/vt/vttablet/tabletmanager/semisyncmonitor/monitor.go

```diff
@@ -300,13 +306,27 @@ func (m *Monitor) startWrites() {
 
 	// Check if we need to continue writing or not.
 	for m.stillBlocked() {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			// We only need to do another write if there were no other successful
+			// writes and we're indeed still blocked.
+			blocked, err := m.isSemiSyncBlocked()
```

Comment:
Do we need to pass the context down through `isSemiSyncBlocked`? I know we're not very consistent about this, but `isSemiSyncBlocked` performs (readonly) queries against MySQL, which in turn can block for various reasons, and can mean that we overshoot the timeout of `ctx`.

Reply:
ExecuteFetch does not use the context for any query timeout related work. BUT, I was thinking about adding a `SELECT /*+ MAX_EXECUTION_TIME(N) */` comment for this. So let me do that now. Thanks!